src/exchanges/binance-base.js

```diff
@@ -205,15 +205,13 @@ class BinanceBase extends BasicClient {
     }
 
     // ticker
-    if (msg.stream === "!ticker@arr") {
-      for (let raw of msg.data) {
-        let remote_id = raw.s;
-        let market = this._tickerSubs.get(remote_id);
-        if (!market) continue;
-
-        let ticker = this._constructTicker(raw, market);
-        this.emit("ticker", ticker, market);
-      }
+    if (msg.stream.toLowerCase().endsWith("ticker")) {
```

Comment:
Add a condition to make sure `msg.stream` is defined first. Otherwise this will throw if the `stream` property doesn't exist for some message.

Reply:
this should be fine, you added a `if (!msg.stream)` before reading value from it

Reply:
ah indeed!